app/services/pinned-entity.service.ts

```diff
@@ -72,20 +73,22 @@ export class PinnedEntityService {
         if (!this.isFavorite(entity)) {
             return;
         }
-
         const url = this._fudgeArmUrl(entity);
         const newFavorites = this._favorites.getValue().filter(pinned => pinned.url !== url);
         this._favorites.next(List<PinnableEntity>(newFavorites));
         this._saveAccountFavorites();
     }
 
-    public getEntityType(entity: NavigableRecord): PinnedEntityType {
+    public getEntityType(entity: NavigableRecord | PinnableEntity): PinnedEntityType {
```

Comment:
i'm not sure this will work if entity is a `PinnableEntity` it is just a normal object. `PinnableEntity` is an interface and that method probably won't work for it.

Reply:
yeah i did wonder about that. It does appear to work fine though. The issue we encountered was that the entity from the drop down was a PinnableEntity, not a NavigableRecord. As NavigableRecords are only from the lists.

without this i wasn't able to check pinnable.pinnableType as that doesn't exist on a NavigableRecord. The isFavorite was failing as the type check returned null.

Reply:
yeah so it works because of what you did at the bottom. PinnableEntity already has the type as a property. I think you should remove it and use that directly
It doesn't make sense here
```
return pinnable ? pinnable.pinnableType : null;
```